lib/js-api/src/externals/table.rs

```diff
@@ -0,0 +1,167 @@
+    ///
+    /// This function will construct the `Table` using the store
+    /// [`BaseTunables`][crate::tunables::BaseTunables].
+    pub fn new(store: &Store, ty: TableType, init: Val) -> Result<Self, RuntimeError> {
+        let descriptor = js_sys::Object::new();
+        js_sys::Reflect::set(&descriptor, &"initial".into(), &ty.minimum.into())?;
+        if let Some(max) = ty.maximum {
+            js_sys::Reflect::set(&descriptor, &"maximum".into(), &max.into())?;
+        }
+        js_sys::Reflect::set(&descriptor, &"element".into(), &"anyfunc".into())?;
+
+        let js_table = js_sys::WebAssembly::Table::new(&descriptor)?;
+        let table = VMTable::new(js_table, ty);
+
+        let num_elements = table.table.length();
+        let func = get_function(init)?;
+        for i in 0..num_elements {
+            set_table_item(&table, i, &func)?;
+        }
+
+        Ok(Self {
+            store: store.clone(),
+            vm_table: table,
+        })
+    }
+
+    /// Returns the [`TableType`] of the `Table`.
+    pub fn ty(&self) -> &TableType {
+        &self.vm_table.ty
+    }
+
+    /// Returns the [`Store`] where the `Table` belongs.
+    pub fn store(&self) -> &Store {
+        &self.store
+    }
+
+    /// Retrieves an element of the table at the provided `index`.
+    pub fn get(&self, index: u32) -> Option<Val> {
+        let func = self.vm_table.table.get(index).ok()?;
+        let ty = FunctionType::new(vec![], vec![]);
+        Some(Val::FuncRef(Some(WasmerFunction::from_vm_export(
+            &self.store,
+            VMFunction::new(func, ty, None),
+        ))))
+    }
+
+    /// Sets an element `val` in the Table at the provided `index`.
+    pub fn set(&self, index: u32, val: Val) -> Result<(), RuntimeError> {
+        let func = get_function(val)?;
+        set_table_item(&self.vm_table, index, &func)?;
+        Ok(())
+    }
+
+    /// Retrieves the size of the `Table` (in elements)
+    pub fn size(&self) -> u32 {
+        self.vm_table.table.length()
+    }
+
+    /// Grows the size of the `Table` by `delta`, initializating
+    /// the elements with the provided `init` value.
+    ///
+    /// It returns the previous size of the `Table` in case is able
+    /// to grow the Table successfully.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the `delta` is out of bounds for the table.
+    pub fn grow(&self, _delta: u32, _init: Val) -> Result<u32, RuntimeError> {
+        unimplemented!();
+    }
+
+    /// Copies the `len` elements of `src_table` starting at `src_index`
+    /// to the destination table `dst_table` at index `dst_index`.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the range is out of bounds of either the source or
+    /// destination tables.
+    pub fn copy(
+        _dst_table: &Self,
+        _dst_index: u32,
+        _src_table: &Self,
+        _src_index: u32,
+        _len: u32,
+    ) -> Result<(), RuntimeError> {
+        unimplemented!();
```

Comment:
Add a message, or remove the `copy` function entirely.

Reply:
Fixed